_docs/dependency-injection.md

```diff
@@ -49,17 +49,17 @@ class AStatelessSessionBean
 }
 ```
 
-After having registered your beans, what is pretty straigthforward when using annotations, you are ready to inject them.
+After having registered your beans, a pretty straigthforward process when using annotations, you are ready to inject them.
 
 ## How to inject an instance
 
-DI can be a manual process where you `inject` an instance, needed by another class, by passing it to the constructor, for example. Inside the `Persistence-Container`, the injection is a process you cannot pursue. Thus, it seems like magic happening behind the scenes. So, instead of manually passing the necessary instances to a classes constructor, the DI container will do that for you.
+DI can be a manual process, where you `inject` an instance needed by another class, by passing it to the constructor, for example. Inside the `Persistence-Container`, the injection is a process, which is actually hard to follow. Thus, it seems like magic happening behind the scenes. So, instead of manually passing the necessary instances to a class' constructor, the DI container will do that for you.
 
-A developer simply has to tell the DI container what instance has to be injected at runtime. The following section describes the two otpions he has.
+A developer simply has to tell the DI container what instance needs to be injected at runtime. The following section describes the two available options.
 
 ### Property Injection
 
-The first option is to annotate a class property using the `@EnterpriseBean` annotation. The annotation accepts a `name` attribute that allows you to specify the name of a bean you have registered before. The following example shows you how to annotate a class property and initiate appserver.io to inject an instance of `AStatelessSessionBean` at runtime.
+The first option is to annotate a class property using the `@EnterpriseBean` annotation. The annotation accepts a `name` attribute, which allows you to specify the name of a bean you have registered. The following example shows you how to annotate a class property and initiate appserver.io to inject an instance of `AStatelessSessionBean` at runtime.
```

Comment:
Is the "before" grammatically incorrect? Beans can only be injected if they have been already registered. That is what the "before" should stress.